### PR TITLE
tests: xtimer_msg_receive_timeout: initialize values

### LINKED ISSUE #4372: xtimer: valgrind complains about uninitialized variables for jumps
Running `tests/xtimer_msg_receive_timeout` in valgrind yields the following result:

```
main(): This is RIOT! (Version: 2015.12-devel-718-gc143-beutlin-pkg/port/lwip)
==7035== Conditional jump or move depends on uninitialised value(s)
==7035==    at 0x804BF92: _is_set (sys/xtimer/xtimer_core.c:53)
==7035==    by 0x804C398: xtimer_remove (sys/xtimer/xtimer_core.c:244)
==7035==    by 0x804C114: xtimer_set (sys/xtimer/xtimer_core.c:122)
==7035==    by 0x804BCB6: xtimer_set_msg (sys/xtimer/xtimer.c:124)
==7035==    by 0x804B99D: main (tests/xtimer_msg_receive_timeout/main.c:34)
==7035==  Uninitialised value was created by a stack allocation
==7035==    at 0x804B949: main (tests/xtimer_msg_receive_timeout/main.c:27)
==7035== 
==7035== Conditional jump or move depends on uninitialised value(s)
==7035==    at 0x804BF9C: _is_set (sys/xtimer/xtimer_core.c:53)
==7035==    by 0x804C398: xtimer_remove (sys/xtimer/xtimer_core.c:244)
==7035==    by 0x804C114: xtimer_set (sys/xtimer/xtimer_core.c:122)
==7035==    by 0x804BCB6: xtimer_set_msg (sys/xtimer/xtimer.c:124)
==7035==    by 0x804B99D: main (tests/xtimer_msg_receive_timeout/main.c:34)
==7035==  Uninitialised value was created by a stack allocation
==7035==    at 0x804B949: main (tests/xtimer_msg_receive_timeout/main.c:27)
==7035== 
Message received: 44
Timeout!
Message received: 44
Timeout!
```

This is due to `target` and/or `long_target` being checked early on an `xtimer_remove()` in `xtimer_set()` and the `xtimer` not being used yet. It works for now because the timer is set in the BSS section, which results in the timer always being 0 in the beginning, but it might lead to unwanted behavior.

### PR DESCRIPTION
**Pull request by authmillenon**
Fixes #4372

### CODE REVIEW

**Comment by haukepetersen** [comment]
Looks valid to me, but was there anything that needed also to be documented, like some timer members have to be initialized before usage or so?

**Comment by authmillenon** [comment]
Any problem if I amend it to this PR?

**Comment by haukepetersen** [comment]
not at all

**Comment by haukepetersen** [comment]
just making sure we don't forget anything that came up in the discussion in #4372

**Comment by authmillenon** [comment]
Provided the docfix.

**Comment by haukepetersen** [comment]
i am good with this, @kaspar030: could you please verify!

File: sys/include/xtimer.h [hunk 1]
```diff
@@ -180,7 +180,9 @@ void xtimer_usleep_until(uint32_t *last_wakeup, uint32_t usecs);
  * The mesage struct specified by msg parameter will not be copied, e.g., it
  * needs to point to valid memory until the message has been delivered.
  *
- * @param[in] timer         timer struct to work with
+ * @param[in] timer         timer struct to work with.
+ *                          Its xtimer_t::target and xtimer_t::long_target
```
Comment (kaspar030) [inline sys/include/xtimer.h]: ```::``` !? ;)

**Comment by kaspar030** [comment]
Or we just always initialize to zero for those functions?

File: sys/include/xtimer.h (hunk 1 above)
Comment (authmillenon) [inline sys/include/xtimer.h]: Auto-generates references to the fields ;-)

**Comment by authmillenon** [comment]
Didn't we came to the conclusion that auto-delete wouldn't work then?

**Comment by kaspar030** [comment]
Yes. Let's go with docs for now. ACK.

**Pull request merged**